Sources/_SwiftSyntaxMacros/ExpressionMacro.swift

```diff
@@ -19,11 +19,12 @@ import _CompilerPluginSupport
 
 /// Describes a macro that is explicitly expanded as an expression.
 public protocol ExpressionMacro: Macro {
-  /// Evaluate a macro described by the given macro expansion expression
+  /// Expand a macro described by the given macro expansion expression
   /// within the given context to produce a replacement expression.
-  static func apply(
-    _ macro: MacroExpansionExprSyntax, in context: MacroEvaluationContext
-  ) -> MacroResult<ExprSyntax>
+  static func expand(
```

Comment:
IMO the method should start with a noun phrase to be in line with the API guidelines, e.g. `expansion(for:in:)`, because the method is not meant to have side effects.

Reply:
A later commit introduces the potential for side effects in the macro evaluation context, e.g., for emitting diagnostics and creating new unique local names.

Reply:
I've thought about this more, and you're totally right. PR incoming to call it `expansion(for:in:)`.